core/ingest/src/main/java/mil/nga/giat/geowave/core/ingest/kafka/IngestFromKafkaDriver.java

```diff
@@ -0,0 +1,346 @@
+package mil.nga.giat.geowave.core.ingest.kafka;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import kafka.consumer.Consumer;
+import kafka.consumer.ConsumerConfig;
+import kafka.consumer.ConsumerIterator;
+import kafka.consumer.KafkaStream;
+import kafka.javaapi.consumer.ConsumerConnector;
+import mil.nga.giat.geowave.core.ingest.AbstractIngestCommandLineDriver;
+import mil.nga.giat.geowave.core.ingest.GeoWaveData;
+import mil.nga.giat.geowave.core.ingest.IngestCommandLineOptions;
+import mil.nga.giat.geowave.core.ingest.IngestFormatPluginProviderSpi;
+import mil.nga.giat.geowave.core.ingest.IngestPluginBase;
+import mil.nga.giat.geowave.core.ingest.avro.AvroFormatPlugin;
+import mil.nga.giat.geowave.core.ingest.avro.GenericAvroSerializer;
+import mil.nga.giat.geowave.core.ingest.local.IngestRunData;
+import mil.nga.giat.geowave.core.store.CloseableIterator;
+import mil.nga.giat.geowave.core.store.DataStore;
+import mil.nga.giat.geowave.core.store.IndexWriter;
+import mil.nga.giat.geowave.core.store.adapter.WritableDataAdapter;
+import mil.nga.giat.geowave.core.store.index.Index;
+import mil.nga.giat.geowave.datastore.accumulo.AccumuloCommandLineOptions;
+import mil.nga.giat.geowave.datastore.accumulo.AccumuloDataStore;
+import mil.nga.giat.geowave.datastore.accumulo.AccumuloOperations;
+
+import org.apache.accumulo.core.client.AccumuloException;
+import org.apache.accumulo.core.client.AccumuloSecurityException;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.apache.log4j.Logger;
+
+/**
+ * /** This class executes the ingestion of intermediate data from a Kafka topic
+ * into GeoWave.
+ * 
+ * @param <I>
+ *            The type for the input data
+ * @param <O>
+ *            The type that represents each data entry being ingested
+ */
+public class IngestFromKafkaDriver<I, O> extends
+		AbstractIngestCommandLineDriver
+{
+	private final static Logger LOGGER = Logger.getLogger(IngestFromKafkaDriver.class);
+	private static int NUM_CONCURRENT_CONSUMERS = KafkaCommandLineOptions.DEFAULT_NUM_CONCURRENT_CONSUMERS;
+	private static ArrayList<String> queue = new ArrayList<String>();
+
+	private KafkaCommandLineOptions kafkaOptions;
+	private AccumuloCommandLineOptions accumuloOptions;
+	private IngestCommandLineOptions ingestOptions;
+	// private final GenericAvroSerializer<I> deserializer;
```

Comment:
cleanup (this and line 69 + avropluginbase comments, really only because you have another commit coming anyways - I think you should delete IngestFromKafkaPlugin, right?)